src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CommitManager.cs

```diff
@@ -194,22 +191,37 @@ internal CommitManager(ITextView textView, RecentItemsManager recentItemsManager
             {
                 // We are on the wrong thread.
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer did not provide Edit Access"));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
             if (subjectBuffer.EditInProgress)
             {
                 FatalError.ReportWithoutCrash(new InvalidOperationException("Subject buffer is editing by someone else."));
-                return AsyncCompletionData.CommitBehavior.None;
+                return new AsyncCompletionData.CommitResult(isHandled: true, AsyncCompletionData.CommitBehavior.None);
             }
 
+            CompletionChange change;
+
+            // We met an issue when external code threw an OperationCanceledException and the cancellationToken is not cancelled.
+            // Catching this scenario for further investigations.
+            // See https://github.com/dotnet/roslyn/issues/38455.
+            try
+            {
+                change = completionService.GetChangeAsync(document, roslynItem, completionListSpan, commitCharacter, cancellationToken).WaitAndGetResult(cancellationToken);
+            }
+            catch (OperationCanceledException e) when (e.CancellationToken != cancellationToken && FatalError.ReportWithoutCrash(e))
```

Comment:
Someone's not combining tokens are they? I've made that mistake before...

Reply:
We no longer check for cancellation token identity. If any form of cancellation occurs after cancellation is requested, we treat it as an expected cancellation.

https://github.com/dotnet/roslyn/blob/6115a5060b0c64c1ace26b838165bfc0a9835288/src/Compilers/Core/Portable/InternalUtilities/FatalError.cs#L94-L95